services/comm/app/operations/message_send.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class MessageSend < Ros::ActivityBase
+  step :retrieve_message
+  failed :message_not_found
+  step :send_message
+  step :update_message_provider
+
+  def retrieve_message(ctx, id:, **)
+    ctx[:message] = Message.find(id)
```

Comment:
you can use `find_by(id: id)` which will return a nil. Nil is a falsy value therefore the failed track kicks in